### ISSUE TITLE
Exception in egmPhotonPUPPIIsolation in several HI workflows

### ISSUE THREAD

**Issue by silviodonato** [cms-sw/cmssw#31894]
In `CMSSW_11_2_X_2020-10-21-2300`, workflows
```
159.1 QCD_Pt_80_120_14_HI_2021+QCD_Pt_80_120_14_HI_2021+DIGIHI2021PPRECO+RECOHI2021PPRECO+HARVESTHI2021PPRECO
159.4 ZEE_14_HI_2021+ZEE_14_HI_2021+DIGIHI2021PPRECO+RECOHI2021PPRECO+HARVESTHI2021PPRECO
310.0 Pyquen_GammaJet_pt20_2760GeV_2021+Pyquen_GammaJet_pt20_2760GeV_2021+DIGIHI2021MIX+RECOHI2021MIX+HARVESTHI2021PPRECO
311.0 Pyquen_DiJet_pt80to120_2760GeV_2021+Pyquen_DiJet_pt80to120_2760GeV_2021+DIGIHI2021MIX+RECOHI2021MIX+HARVESTHI2021PPRECO
312.0 Pyquen_ZeemumuJets_pt10_2760GeV_2021+Pyquen_ZeemumuJets_pt10_2760GeV_2021+DIGIHI2021MIX+RECOHI2021MIX+HARVESTHI2021PPRECO
``` 
throw an exception 
```
----- Begin Fatal Exception 22-Oct-2020 05:07:21 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 3
   [1] Running path 'dqmofflineOnPAT_1_step'
   [2] Prefetching for module SingleTopTChannelLeptonDQM_miniAOD/'singleTopElectronMediumDQM_miniAOD'
   [3] Prefetching for module PATMETSlimmer/'slimmedMETs'
   [4] Prefetching for module PATMETProducer/'patMETs'
   [5] Prefetching for module PFEGammaToCandidate/'pfeGammaToCandidate'
   [6] Prefetching for module PATPhotonSelector/'selectedPatPhotons'
   [7] Prefetching for module PATPhotonProducer/'patPhotons'
   [8] Calling method for module CITKPFIsolationSumProducerForPUPPI/'egmPhotonPUPPIIsolation'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: edm::ValueMap<float>
Looking for module label: puppi
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```
Since all crashing workflow are related to HI, the error seems related to #31668

**Comment by silviodonato** [comment]
assign reconstruction

**Comment by silviodonato** [comment]
@mandrenguyen

**Comment by mandrenguyen** [comment]
@silviodonato Checking now

**Comment by silviodonato** [comment]
fix in #31897
